subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/BomSupportPluginsSmokeTest.groovy

```diff
@@ -0,0 +1,383 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.smoketests
+
+import org.gradle.integtests.fixtures.resolve.ResolveTestFixture
+import org.gradle.test.fixtures.file.TestFile
+import spock.lang.Unroll
+
+class BomSupportPluginsSmokeTest extends AbstractSmokeTest {
+    static bomVersion = "1.5.8.RELEASE"
+    static bom = "'org.springframework.boot:spring-boot-dependencies:$bomVersion'"
+
+    @Unroll
+    def 'bom support is provided by #bomSupportProvider'() {
+        given:
+        testProjectDir.newFile('settings.gradle') << """
+            rootProject.name = 'springbootproject'
+        """
+        def buildScript = """
+            plugins {
+                id "java"
+                ${dependencyManagementPlugin}
+            }
+            ${jcenterRepository()}
+
+            ${bomDeclaration}
+
+            dependencies {
+                implementation "org.springframework.boot:spring-boot"
+                implementation "org.springframework.boot:spring-boot-autoconfigure"
+            
+                testImplementation "junit:junit"
+                testImplementation "org.springframework:spring-test"
+                testImplementation "org.springframework.boot:spring-boot-test"
+                testImplementation "org.springframework.boot:spring-boot-test-autoconfigure"
+            }
+        """
+        def resolve = new ResolveTestFixture(new TestFile(buildFile), 'testCompileClasspath')
+        resolve.prepare(buildScript)
+
+        when:
+        runner('checkDep').build()
+
+        then:
+        def expected = expectedDependencyTree(reason1, reason2, reason3)
+        def actual = resolve.getResultFile().text.split('\n')
+        actual.size() == actual.size()
+        actual.each {
+            assert expected.contains(it)
+        }
+
+        where:
+        bomSupportProvider                    | reason1          | reason2          | reason3          | bomDeclaration                                        | dependencyManagementPlugin
+        "gradle"                              | "conflict"       | "conflict"       | "requested"      | "dependencies { implementation $bom }"                |""
+        "nebula recommender plugin"           | "selectedByRule" | "requested"      | "requested"      | "dependencyRecommendations { mavenBom module: $bom }" |"id 'nebula.dependency-recommender' version '5.0.0'"
+        "spring dependency management plugin" | "selectedByRule" | "selectedByRule" | "selectedByRule" | "dependencyManagement { imports { mavenBom $bom } }"  |"id 'io.spring.dependency-management' version '1.0.3.RELEASE'"
+
+    }
+
+    private static expectedDependencyTree(String reason1, String reason2, String reason3) {
```

Comment:
⭕️ Wouldn't it be better to assert that the trees are the same instead of checking in a point in time? That would make it easier to upgrade this test in the future.

Reply:
I was starting to do it like that but it was hard to figure out the different variants. However, now that I had the test running, I gave it another try and succeeded.
It took me a bit, but it's better to understand and maintain now I think. Thanks for pushing. :)

Reply:
It still asserts a certain point in time, though a lot more readable. What I expected was the trees to all just be equal, but maybe that's not possible.